src/Generators/MigrationGenerator.php

```diff
@@ -294,6 +294,9 @@ protected function buildForeignKey(string $column_name, ?string $on, string $typ
         } elseif (Str::contains($on, '.')) {
             [$table, $column] = explode('.', $on);
             $table = Str::snake($table);
+        } elseif (Str::contains($on, '\\')) {
+            $table = (new $on())->getTable();
```

Comment:
Depending on the order of operations (i.e. migrations generated before models), it is possible this model will not exist and as such this dynamic creation will fail.

Reply:
Thought about that as well, first had a `class_exists` check. Maybe if the class does not exists, we can fallback to a plural, lowercase version of the model? E.g. `\Foo\Bar\Attachment` becomes `attachments`,

Reply:
To limit the code paths, I would not use the dynamic call and just do the plural, lowercase name of the model. I believe this is done elsewhere in the code.

Reply:
Updated!